### DISCUSSION TITLE
Maximum wait time during TCP closing handshake (FIN)?

### DISCUSSION THREAD

**Discussion by michaelnwani** [uNetworking/uWebSockets.js#240] [Q&A] [answered]
Hi Alex,

Given a client with a spotty wifi connection/bad network conditions, is there a maximum amount of time that websockets might wait for the client to respond during a (graceful) closing handshake? This would be after calling `ws.end(<code>, <msg>)`. I hope this question makes sense.

Thanks!

**Comment by ghost** [comment, marked as answer]
Yes the timeout is the same as idleTimeout

**Comment by michaelnwani** [comment]
@alexhultman thanks! That resolves my issue
